server.js

```diff
@@ -9,14 +11,57 @@ let fruits = [
   { id: "5", name: 'Orange', color: 'orange'}
 ]
 
+// @ GET - Display all
 app.get('/fruits', (req, res) => {
-  res.json(fruits)
+  const sortFruits = fruits.sort((a,b)=> a.id-b.id)
+  res.json(sortFruits)
 });
 
+// @ GET single fruit
 app.get('/fruits/:id', (req, res) => {
-  res.json({})
+  const {id} = req.params;
+  const findFruit = fruits.find(item=> item.id === id);
+  res.json(findFruit)
+})
+
+// @ POST - Add new fruit
+app.post("/fruits/add", (req,res)=>{
```

Comment:
In routes for adding new data, typically we do not append /add to the route.

/add we would just would use in case we serve a frontend form from the backend. So actually we would just it for a GET route, GET /fruits/add in this case.